austrakka/components/sequence/sync/sync_workflow.py

```diff
@@ -308,19 +309,26 @@ def detect_and_record_obsolete_files(int_med, sync_state):
         DETECTION_DATE_KEY: []
     })
 
-    for (
-            root_dir,
-            dir_names,
-            file_names) in os.walk(
-            get_output_dir(sync_state)):
-        dir_names[:] = [d for d in dir_names if d not in [TRASH_DIR]]
-        for name in file_names:
-            if os.path.splitext(name)[-1] in [FASTQ_EXT, FASTA_EXT, GZ_EXT]:
-                files_on_disk.append((
-                    os.path.join(root_dir, name),
-                    name,
-                    datetime.now().strftime("%Y-%m-%d %H:%M:%S.%f")
-                ))
+    sample_subdirectories = [
+        f.path for f in os.scandir(
+            get_output_dir(sync_state)) if f.is_dir()]
+    files = sum([[f.path for f in os.scandir(subdir) if not f.is_dir()]
```

Comment:
It looks like this will scan just one level deep. In the purging stage, the code will walk through output_dir and delete all  empty directories. What do you think should be done there? The initial build of this was modeled on our usage of the cli for apg where the directory is read-only to users and we leave the directory in a pristine state.